src/main/scala/dissonance/model/message/Update.scala

```diff
@@ -0,0 +1,24 @@
+package dissonance.model.message
+
+import cats.implicits._
+import dissonance.model.embed.Embed
+import dissonance.model.Snowflake
+import io.circe.Decoder
+import io.circe.generic.extras.Configuration
+import io.circe.generic.extras.semiauto._
+
+sealed trait Update extends Product with Serializable
+
+object Update {
+  case class Edit(message: Message) extends Update
+  // Every time an embed is created, this event is generated immediately after the MESSAGE_CREATE
+  case class EmbedUpdate(
+      id: Snowflake,
+      channelId: Snowflake,
+      guildId: Snowflake,
+      embeds: List[Embed]
+  ) extends Update
+
+  implicit val config: Configuration          = Configuration.default.withSnakeCaseMemberNames
+  implicit val updateDecoder: Decoder[Update] = Decoder[Message].map(Edit).widen[Update] or deriveConfiguredDecoder[EmbedUpdate].widen[Update]
```

Comment:
Interesting. So widen basically casts to a supertype?
Also, where does this `or` come from? Is that from circe?

Reply:
Ya I actually want to understand it better because I forgot what makes it _legal_ , remind me to ask in Gitter tomorrow. Ya the `or` comes from Circe 👍